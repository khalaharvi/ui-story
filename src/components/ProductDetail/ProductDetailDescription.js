import React from 'react';
import PropTypes from 'prop-types';

import { cx } from './ProductDetail';

const ProductDetailDescription = ({ description }) => (
  <div
    itemProp="description"
    dangerouslySetInnerHTML={{ __html: description }}
    className={cx('description')}
  />
);

ProductDetailDescription.propTypes = {
  description: PropTypes.string.isRequired
};

export default ProductDetailDescription;
