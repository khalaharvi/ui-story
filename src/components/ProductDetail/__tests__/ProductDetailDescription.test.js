import React from 'react';
import { shallow } from 'enzyme';

import ProductDetailDescription from '../ProductDetailDescription';

describe('<ProductDetailDescription />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProductDetailDescription
        description="Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eum quisquam numquam eos, porro illum sapiente molestiae. In voluptas aspernatur consequatur beatae, reiciendis ut quidem voluptatum omnis laudantium? Amet, ut, earum!"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with HTML description', () => {
    const wrapper = shallow(
      <ProductDetailDescription
        description={`
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quis quaerat adipisci voluptatibus. Tempora quas at quasi quae optio laudantium, modi suscipit quidem atque aliquid earum veritatis voluptatum unde harum, repudiandae!</p>
        `}
      />
    );

    expect(wrapper.render().find('p')).toHaveLength(1);
  });
});
