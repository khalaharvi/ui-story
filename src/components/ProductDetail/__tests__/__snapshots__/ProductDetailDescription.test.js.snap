// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailDescription /> renders correctly 1`] = `
<div
  className="dh-product-detail-description"
  dangerouslySetInnerHTML={
    Object {
      "__html": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Eum quisquam numquam eos, porro illum sapiente molestiae. In voluptas aspernatur consequatur beatae, reiciendis ut quidem voluptatum omnis laudantium? Amet, ut, earum!",
    }
  }
  itemProp="description"
/>
`;
